src/rust/engine/run-all-tests.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash -u
+
+# This script exists to work around https://github.com/rust-lang/rust/issues/44862
+# `cargo test --all` doesn't work because we can't use a workspace, so this script replicates its
+# behavior.
+
+here=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
+
+exit_code=0
+
+for crate in $(find ${here} -name Cargo.toml); do
+  (
+    dir=$(dirname ${crate})
+    echo >&2 "Running tests in ${dir}:"
+    cd ${dir} && cargo test
```

Comment:
This is not going to be the right cargo, on my machine it'd pick up nightly for example. Seems like sourcing `bootstrap.sh` etc. is needed.

Reply:
Done